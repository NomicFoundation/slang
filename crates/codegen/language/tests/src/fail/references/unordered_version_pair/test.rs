#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(Language(
    name = Foo,
    documentation_dir = "foo/bar",
    root_item = One,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0", "2.0.0", "3.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                Struct(
                    name = One,
                    fields = (
                        field_1 = Optional(
                            reference = Two,
                            enabled = Range(from = "3.0.0", till = "2.0.0")
                        ),
                        field_2 = Optional(reference = Two)
                    )
                ),
                Token(
                    name = Two,
                    definitions = [TokenDefinition(scanner = Atom("two"))]
                )
            ]
        )]
    )],
    queries = ()
));

fn main() {}
