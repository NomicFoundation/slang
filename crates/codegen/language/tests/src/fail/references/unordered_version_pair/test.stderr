error: Version '3.0.0' must be less than corresponding version '2.0.0'.
  --> src/fail/references/unordered_version_pair/test.rs:19:42
   |
19 | ...                   enabled_in = "3.0.0",
   |                                    ^^^^^^^
